abba/json/src/Parse.cpp

```diff
@@ -0,0 +1,480 @@
+#include "Parse.hpp"
+
+Parse::Parse(std::string filename) : json(filename)
+{
+
+}
+
+void Parse::parse()
+{
+	_file = &json.getFile();
+	std::cout << "i'm here" << std::endl;
+	getToken();
+	Json *root = new Json();
+	root->type = Json::OBJECT;
+	root = parseObject(root);
+	std::cout << "DONE PARSING!!!!!!!!!!!!" << std::endl;
+}
+
+char Parse::getWithoutWhiteSpace()
+{
+	std::cout << "getWithoutWhiteSpace function called" << std::endl;
+	char	c = ' ';
+	while (c == ' ' || c == '\t' || c == '\n' || \
+			c == '\v' || c == '\f' || c == '\r')
+	{
+		_file->get(c);
+		if (!_file)
+			return c;
+	}
+	return c;
+}
+
+Parse::Type Parse::getToken()
+{
+	char		c;
+	Type	type;
+
+//	std::cout << "getToken function called" << std::endl;
+	if (_file->eof())
+	{
+		std::cout << "finish reading" << std::endl;
+	}
+	_prevPos = _file->tellg();
+	c = getWithoutWhiteSpace();
+	if (c == '"')
+	{
+		type.type = STRING;
+		_file->get(c);
+		while (c != '\n' && c != '"')
+		{
+			type.value += c;
+			_file->get(c);
+		}
+	}
+	else if (c == '[')
+	{
+//		std::cout << "array open" << std::endl;
+		type.type = ARRAY_OPEN;
+	}
+	else if (c == ']')
+	{
+//		std::cout << "array close" << std::endl;
+		type.type = ARRAY_CLOSE;
+	}
+	else if (c == '{')
+	{
+//		std::cout << "curly open" << std::endl;
+		type.type = CURLY_OPEN;
+	}
+	else if (c == '}')
+		type.type = CURLY_CLOSE;
+	else if (c == ':')
+		type.type = COLON;
+	else if (c == ',')
+		type.type = COMMA;
+	else if (c >= '0' && c <= '9')
+	{
+		type.type = NUMBER;
+		while (c >= '0' && c <= '9')
+		{
+			_prevPos = _file->tellg();
+			type.value += c;
+			_file->get(c);
+		}
+		_file->seekg(_prevPos);
```

Comment:
Instead of constantly changing the file position using seek you can use `peek()` instead of `get()` so that the position won't be forwarded in the first place, that also means you won't have to keep track of a previous position.

Reply:
so that means I should always use peek and if it is what I'm looking for I use get, right?

Reply:
if you have some conditional get based on the next character then yeah